lib/Controller/Api1Controller.php

```diff
@@ -196,6 +196,33 @@ public function updateTable(int $tableId, string $title = null, string $emoji =
 		}
 	}
 
+	/**
+	 * Transfer table from one user to another
+	 *
+	 * @NoAdminRequired
+	 * @CORS
+	 * @NoCSRFRequired
+	 *
+	 * @param int $tableId Table ID
+	 * @param string $newOwnerUserId New user ID 
+	 * @param string $userId Current user ID 
+	 * @return DataResponse<Http::STATUS_OK, TablesTable, array{}>|DataResponse<Http::STATUS_FORBIDDEN|Http::STATUS_INTERNAL_SERVER_ERROR|Http::STATUS_NOT_FOUND, array{message: string}, array{}>
+	 *
+	 */
+	public function transferTable(int $tableId, int $newOwnerUserId, int $userId): DataResponse {
+		try {
+			return new DataResponse($this->tableService->transfer($tableId, $newOwnerUserId, $userId)->jsonSerialize());
+		} catch (PermissionError $e) {
+			$this->logger->warning('A permission error occurred: ' . $e->getMessage());
+			$message = ['message' => $e->getMessage()];
+			return new DataResponse($message, Http::STATUS_FORBIDDEN);
+		} catch (InternalError|Exception $e) {
+			$this->logger->warning('An internal error or exception occurred: '.$e->getMessage());
+			$message = ['message' => $e->getMessage()];
+			return new DataResponse($message, Http::STATUS_INTERNAL_SERVER_ERROR);
+		}
+	}
+
```

Comment:
This is how I envision the actual "transfer ownership" functionality to be done. But I have no idea how to connect this to the frontend. Or if this is the right way to go about it.